### PR TITLE
Add DFN-4

### PR DESCRIPTION
**Pull request by DanSGiesbrecht**
Used for Sensiron SHTC3

---

image

image

---

From [datasheet](https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/0_Datasheets/Humidity/Sensirion_Humidity_Sensors_SHTC3_Datasheet.pdf):

image

image

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -333,7 +368,7 @@ DFN-8-1EP_4x4mm_P0.8mm_EP2.3x3.24mm:
 
   EP_size_x_overwrite: 2.30
   EP_size_y_overwrite: 3.24
-
```
Comment (DanSGiesbrecht) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml]: Save automatically removed whitespace.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -696,5 +731,3 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
   suffix: ''
   #include_suffix_in_3dpath: 'False'
   #include_pad_size: 'fp_name_only' # 'both' | 'none' (default)
-
```
Comment (DanSGiesbrecht) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml]: Save automatically removed extra lines at end.

**Comment by poeschlr** [comment]
I am not really happy about the available documentation of this part. Is there possibly some other document available that contains tolerance information? Otherwise, we might be better off to use the suggested footprint as is instead of using the ipc generator.

**Comment by DanSGiesbrecht** [comment]
Thanks for the review, @poeschlr. I agree that there is not great documentation for the package. I cannot find any other official source for package dimensions.

So, do you suggest the following steps?
1) close this PR
2) implement the manufacturer-recommended footprint

If you agree with 1) and 2) above:
- I assume that if it's not IPC-compliant, it should not be a generic `DFN-4` part, but rather `Sensirion_DFN-4...`. Is that correct?
- where should this footprint be located? Should I create a `Sensor_Humidity.pretty`?

**Comment by poeschlr** [comment]
Sensor_Humidity sounds like a good library name. (A humidity sensor will most likely have some opening in the package that could be represented in the 3d model. Another reason for a speialized footprint here)

Prefixing with manufacturer name should be ok

I think there is no harm in having rounded rectangle pads using IPC rules.

**Comment by DanSGiesbrecht** [comment]
Thanks! I'll start work on this. It'll give me a good excuse to learn the 3D CAD workflow!

**Pull request closed without merging**